clang/include/clang/Sema/Sema.h

```diff
@@ -11392,9 +11392,9 @@ class Sema final : public SemaBase {
       CXXScopeSpec &SS, IdentifierInfo *Name, SourceLocation NameLoc,
       const ParsedAttributesView &Attr, TemplateParameterList *TemplateParams,
       AccessSpecifier AS, SourceLocation ModulePrivateLoc,
-      SourceLocation FriendLoc, unsigned NumOuterTemplateParamLists,
-      TemplateParameterList **OuterTemplateParamLists,
-      SkipBodyInfo *SkipBody = nullptr);
+      SourceLocation FriendLoc,
+      ArrayRef<TemplateParameterList *> OuterTemplateParamLists,
+      bool IsMemberSpecialization, SkipBodyInfo *SkipBody = nullptr);
```

Comment:
I have a preference for using an enum instead of bool params (so that callers do: `SpecializationKind::None`, or `SpecializationKind::Member` or something like that?).  Even if there are two options, it improves readability of callers a ton.

Reply:
I still have a preference for something like this, but otherwise